In [18]:
import pandas as pd
import numpy as np
from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
from sklearn.preprocessing import StandardScaler

from sklearn import svm
from sklearn.svm import SVC
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import classification_report


from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.model_selection import train_test_split, cross_val_score

In [19]:
forest=pd.read_csv("forestfires.csv")
forest

month  day  FFMC    DMC     DC   ISI  temp  RH  wind  rain  ...  monthfeb  \
0     mar  fri  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0  ...         0   
1     oct  tue  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0  ...         0   
2     oct  sat  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0  ...         0   
3     mar  fri  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2  ...         0   
4     mar  sun  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0  ...         0   
..    ...  ...   ...    ...    ...   ...   ...  ..   ...   ...  ...       ...   
512   aug  sun  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0  ...         0   
513   aug  sun  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  ...         0   
514   aug  sun  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  ...         0   
515   aug  sat  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0  ...         0   
516   nov  tue  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0  ...         0   

     monthjan  monthjul  monthjun  monthmar  monthmay  monthnov  monthoct  \
0           0         0         0         1         0         0         0   
1           0         0         0         0         0         0         1   
2           0         0         0         0         0         0         1   
3           0         0         0         1         0         0         0   
4           0         0         0         1         0         0         0   
..        ...       ...       ...       ...       ...       ...       ...   
512         0         0         0         0         0         0         0   
513         0         0         0         0         0         0         0   
514         0         0         0         0         0         0         0   
515         0         0         0         0         0         0         0   
516         0         0         0         0         0         1         0   

     monthsep  size_category  
0           0          small  
1           0          small  
2           0          small  
3           0          small  
4           0          small  
..        ...            ...  
512         0          large  
513         0          large  
514         0          large  
515         0          small  
516         0          small  

[517 rows x 31 columns]

Attribute Information:

X : x-axis spatial coordinate within the Montesinho park map: 1 to 9

Y : y-axis spatial coordinate within the Montesinho park map: 2 to 9

month : month of the year: ‘jan’ to ‘dec’

day : day of the week: ‘mon’ to ‘sun’

FFMC : FFMC (Fine Fuel Moisture Code) index from the FWI system: 18.7 to 96.20

DMC : DMC (Duff Moisture Code) index from the FWI system: 1.1 to 291.3

DC : DC (Drought Code) index from the FWI system: 7.9 to 860.6

ISI : ISI (Initial Spread Index) index from the FWI system: 0.0 to 56.10

temp : temperature in Celsius degrees: 2.2 to 33.30

RH : relative humidity in %: 15.0 to 100

wind : wind speed in km/h: 0.40 to 9.40

rain : outside rain in mm/m2 : 0.0 to 6.4

In [20]:
forest.replace(dict.fromkeys(['mon',"tue",'thu','wed','fri'],"weekdays"), inplace=True)
forest.replace(dict.fromkeys(['sun','sat'],"weekend"), inplace=True)
forest.replace({"small":0,"large":1}, inplace=True)
forest.replace({"weekdays":0,"weekend":1}, inplace=True)
int_variables=['FFMC', 'DMC', 'DC', 'ISI', 'temp', 'RH', 'wind', 'rain', 'area']
forest_n=forest.drop("month", axis=1)

In [21]:
forest_n

day  FFMC    DMC     DC   ISI  temp  RH  wind  rain   area  ...  \
0      0  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00  ...   
1      0  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00  ...   
2      1  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00  ...   
3      0  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00  ...   
4      1  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00  ...   
..   ...   ...    ...    ...   ...   ...  ..   ...   ...    ...  ...   
512    1  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44  ...   
513    1  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29  ...   
514    1  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16  ...   
515    1  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00  ...   
516    0  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00  ...   

     monthfeb  monthjan  monthjul  monthjun  monthmar  monthmay  monthnov  \
0           0         0         0         0         1         0         0   
1           0         0         0         0         0         0         0   
2           0         0         0         0         0         0         0   
3           0         0         0         0         1         0         0   
4           0         0         0         0         1         0         0   
..        ...       ...       ...       ...       ...       ...       ...   
512         0         0         0         0         0         0         0   
513         0         0         0         0         0         0         0   
514         0         0         0         0         0         0         0   
515         0         0         0         0         0         0         0   
516         0         0         0         0         0         0         1   

     monthoct  monthsep  size_category  
0           0         0              0  
1           1         0              0  
2           1         0              0  
3           0         0              0  
4           0         0              0  
..        ...       ...            ...  
512         0         0              1  
513         0         0              1  
514         0         0              1  
515         0         0              0  
516         0         0              0  

[517 rows x 30 columns]

In [22]:
from sklearn.preprocessing import StandardScaler
scaler=StandardScaler()
forest_n[int_variables]=scaler.fit_transform(forest_n[int_variables])

In [23]:
forest_n.head(2)

day      FFMC       DMC        DC       ISI      temp        RH      wind  \
0    0 -0.805959 -1.323326 -1.830477 -0.860946 -1.842640  0.411724  1.498614   
1    0 -0.008102 -1.179541  0.488891 -0.509688 -0.153278 -0.692456 -1.741756   

       rain     area  ...  monthfeb  monthjan  monthjul  monthjun  monthmar  \
0 -0.073268 -0.20202  ...         0         0         0         0         1   
1 -0.073268 -0.20202  ...         0         0         0         0         0   

   monthmay  monthnov  monthoct  monthsep  size_category  
0         0         0         0         0              0  
1         0         0         1         0              0  

[2 rows x 30 columns]

In [24]:
from sklearn.model_selection import GridSearchCV

In [26]:
x=forest_n.drop("size_category",axis=1)
y=forest_n["size_category"]
x_train, x_test, y_train, y_test = train_test_split(x, y)

In [27]:
clf = SVC()
param_grid = [{'kernel':['rbf'],'gamma':[50,5,10,0.5],'C':[15,14,13,12,11,10,0.1,0.001] }]
gsv = GridSearchCV(clf,param_grid,cv=10)
gsv.fit(x_train,y_train)

GridSearchCV(cv=10, estimator=SVC(),
             param_grid=[{'C': [15, 14, 13, 12, 11, 10, 0.1, 0.001],
                          'gamma': [50, 5, 10, 0.5], 'kernel': ['rbf']}])

In [28]:
gsv.best_params_ , gsv.best_score_

({'C': 15, 'gamma': 0.5, 'kernel': 'rbf'}, 0.7675438596491228)

In [29]:
clf = SVC(C= 15, gamma = 0.5, kernel="rbf")
clf.fit(x_train , y_train)
y_pred = clf.predict(x_test)
acc = accuracy_score(y_test, y_pred) * 100
print("Accuracy =", acc)
confusion_matrix(y_test, y_pred)

Accuracy = 75.38461538461539


array([[85,  7],
       [25, 13]], dtype=int64)

In [30]:
clf.score(x_test,y_test)

0.7538461538461538